projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -4800,37 +4726,48 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
      * @memberof IgxGridBaseComponent
      */
     public deselectRows(rowIDs: any[]) {
-        let newSelection: Set<any>;
-        newSelection = this.selection.delete_items(this.id, rowIDs);
-        this.triggerRowSelectionChange(newSelection);
+        this.selectionService.deselectRowsWithNoEvent(rowIDs);
+        this.cdr.markForCheck();
     }
 
     /**
      * Selects all rows
-     * Note: If filtering is in place, selectAllRows() and deselectAllRows() select/deselect all filtered rows.
+     * Note: By default if filtering is in place, selectAllRows() and deselectAllRows() select/deselect all filtered rows.
+     * If you set the parameter onlyFilterData to false that will select all rows in the grid exept deleted rows.
      * ```typescript
      * this.grid.selectAllRows();
+     * this.grid.selectAllRows(false);
      * ```
+     * @param onlyFilterData
 	 * @memberof IgxGridBaseComponent
      */
-    public selectAllRows() {
-        this.triggerRowSelectionChange(this.selection.get_all_ids(this.gridAPI.get_all_data(true), this.primaryKey));
+    public selectAllRows(onlyFilterData = true) {
+        const data = onlyFilterData && this.filteredData ? this.filteredData : this.gridAPI.get_all_data();
+        const rowIDs = this.selectionService.getRowIDs(data).filter(rID => !this.gridAPI.row_deleted_transaction(rID));
+        this.selectRows(rowIDs);
     }
 
     /**
      * Deselects all rows
+     * Note: By default if filtering is in place, selectAllRows() and deselectAllRows() select/deselect all filtered rows.
+     * If you set the parameter onlyFilterData to false that will select all rows in the grid exept deleted rows.
      * ```typescript
      * this.grid.deselectAllRows();
      * ```
-     * Note: If filtering is in place, selectAllRows() and deselectAllRows() select/deselect all filtered rows.
+     * @param onlyFilterData
+	 * @memberof IgxGridBaseComponent
      */
-    public deselectAllRows() {
-        this.triggerRowSelectionChange(this.selection.get_empty());
+    public deselectAllRows(onlyFilterData = true) {
+        if (onlyFilterData && this.filteredData && this.filteredData.length > 0) {
+            this.deselectRows(this.selectionService.getRowIDs(this.filteredData));
+        } else {
+            this.selectionService.selectRowsWithNoEvent([], true);
```

Comment:
Is this an apocryphal way of deselecting everything without throwing events? :)

It's probably better to expose this seperately (isn't ```clear``` doing the same) as it won't be obvious for everyone looking at the code.